src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -647,6 +721,103 @@ public static boolean isDST(String timezone) {
     return !zoneId.getRules().getTransitionRules().isEmpty();
   }
 
+  /**
+   * Read from `sun.util.calendar.ZoneInfo` via reflection to get timezone info.
+   * 
+   * @param tzId timezone id
+   * @return timezone info
+   */
+  private static TzInfoInJavaUtilForORC getInfoForUtilTZ(String tzId) {
+    TimeZone tz = TimeZone.getTimeZone(tzId);
+    if (!(tz instanceof ZoneInfo)) {
+      throw new UnsupportedOperationException("Unsupported timezone: " + tzId);
+    }
+    ZoneInfo zoneInfo = (ZoneInfo) tz;
+
+    // The constructor of TimeZoneInfoInJavaUtilPackage will extract and repack
+    // transitions info.
+    return new TzInfoInJavaUtilForORC(
+        (long[]) FieldUtils.readField(zoneInfo, "transitions"),
+        (int[]) FieldUtils.readField(zoneInfo, "offsets"),
+        (int) FieldUtils.readField(zoneInfo, "rawOffset"));
+  }
+
+  private static ColumnVector getTransitionsForUtilTZ(TzInfoInJavaUtilForORC info) {
+    try (HostColumnVector hcv = HostColumnVector.fromLongs(info.transitions)) {
+      return hcv.copyToDevice();
```

Comment:
Please correct my understanding of the scale here:  I'm guessing the number of transitions per time-zone id is of the order of a few hundreds?